### ISSUE TITLE
Track `menu-bar-open` in TTY only

### ISSUE THREAD

**Issue by lastquestion** [lastquestion/explain-pause-mode#19]
This is elisp, which calls different kinds of menus. Need to test the various cases and decide what to do. 

In TTY, the TTY menu eventually calls `read_key_sequence` in C land, and that time is not tracked, showing up as a slow call.

~- [ ] test graphical mouse and non mouse~ TTY only
- [x] test `tmm-menubar` 
- [x] test `popup-menu` (F10 when menubar is visible)
- [x] figure out what both call down into to find the best place to advise

Optimally, we should treat menu like minibuffer, splitting up the execution time and restarting the tracking when the command begins execution, if possible.

**Comment by lastquestion** [comment]
#42 fixes this for TTY in minibuffer menu bar, but not dropdown menubar.

**Comment by lastquestion** [comment]
Doesn't work in windowed mode, function is `menu-find-file-existing`

**Comment by lastquestion** [comment]
`menu-find-file-existing` ultimately calls `x-file-dialog`, opening a new issue for that later. Focusing this only on TTY.

**Comment by lastquestion** [comment]
Notes:
`F10` -> `popup-menu` -> `x-popup-menu` (called a lot of places), C code -> `menu_show_hook` (C).
* `x-popup-menu` only called in `keyboard.c`, in `read_char_x_menu_prompt` from `read_char` which we already covered in #17 

Splits into 
  -> `x_menu_show`
  -> `ns_menu_show`
  -> `w32_menu_show`
  -> `tty_menu_show`

* Timers definitely run when menu is open. Add a test for this case. Also, what if minibuffer is open and THEN you open menu?

* `tty_menu_show` -> `tty_menu_activate` -> `read_menu_input` -> `read_menu_command` -> `read_key_sequence` -> `read_char` is why

TL;DR we should be safe to advise `x-popup-menu`, on TTY at the very least. Insert a native frame like `read_char` family.

**Comment by lastquestion** [comment]
Timer with minibuffer open THEN F10:
```elisp
;; enter m-x
("enter" "execute-extended-command" "root-emacs" "root-emacs" nil 0 nil nil nil 1)
("enter" "completing-read-default" "execute-extended-command" "execute-extended-command" nil 0 nil nil nil 2)
;; advise read-from-minibuffer
("enter" "#<subr read-from-minibuffer>" "completing-read-default" "completing-read-default" t 0 nil nil nil 3)
;; timer runs
("enter" "silly" "#<subr read-from-minibuffer>" "timer" nil 0 nil nil nil 3)
("exit" "silly" "timer" 0 nil)
("enter" "silly" "#<subr read-from-minibuffer>" "timer" nil 0 nil nil nil 3)
("exit" "silly" "timer" 0 nil)
;; open menu-bar
("enter" "menu-bar-open" "#<subr read-from-minibuffer>" "#<subr read-from-minibuffer>" nil 0 nil nil nil 4)
;; close menu-bar with c-g
("exit" "menu-bar-open" "#<subr read-from-minibuffer>" 98114 profile)
;; close minibuffer with c-g
("enter" "abort-recursive-edit" "#<subr read-from-minibuffer>" "#<subr read-from-minibuffer>" nil 0 nil nil nil 4)
("exit" "abort-recursive-edit" "#<subr read-from-minibuffer>" 0 nil)
("exit" "#<subr read-from-minibuffer>" "completing-read-default" 6035 nil)
("exit" "completing-read-default" "execute-extended-command" 0 nil)
("exit" "execute-extended-command" "root-emacs" 0 nil)
```

**Comment by lastquestion** [comment]
```elisp
;; with advising `x-popup-menu`
("enter" "menu-bar-open" "root-emacs" "root-emacs" nil 0 nil nil nil 1)
("enter" "#<subr x-popup-menu>" "menu-bar-open" "menu-bar-open" t 0 nil nil nil 2)
("exit" "#<subr x-popup-menu>" "menu-bar-open" 5915 nil)
("exit" "menu-bar-open" "root-emacs" 1 nil)
```
